python-runnables/resize-node-pool/runnable.py

```diff
@@ -42,20 +42,45 @@ def run(self, progress_callback):
         node_pool_id = node_pool.name
         logging.info("Node pool selected is %s " % node_pool_id)
 
-        desired_count = self.config['numNodes']
-        logging.info("Resize to %s" % desired_count)
-        if desired_count == 0:
-            if len(node_pools) == 1:
-                raise Exception("Can't delete node pool, a cluster needs at least one running node pool")
-            def do_delete():
-                cluster_update_op = clusters.agent_pools.begin_delete(resource_group, cluster_name, node_pool_id)
-                return cluster_update_op.result()
-            delete_result = run_and_process_cloud_error(do_delete)
-            logging.info("Cluster updated")
-            return '<pre class="debug">Node pool %s deleted</pre>' % node_pool_id
+        autoscaling_enabled = self.config['autoScaling']
+        node_pool.enable_auto_scaling = autoscaling_enabled
+        if autoscaling_enabled:
+            min_nodes = self.config['minNumNodes']
+            max_nodes = self.config['maxNumNodes']
+            initial_node_count = self.config['numNodes']
+            if min_nodes is None:
```

Comment:
nit: If those are mandatory I'm not sure we really need those check (my main issue with those check is the lack of consistence between this part of the if and the else where there is no check on numNodes)